src/monodroid/jni/Android.mk

```diff
@@ -57,14 +60,14 @@ LOCAL_SRC_FILES := \
 	jni.c \
 	monodroid-glue.c \
 	util.c \
-	logger.c \
 	debug.c \
 	timezones.c \
 	zip/ioapi.c \
 	zip/unzip.c \
 	xamarin_getifaddrs.c \
 	cpu-arch-detect.c \
-	monodroid-networkinfo.c
+	monodroid-networkinfo.c \
+	../$(JAVA_INTEROP_PATH)/src/java-interop/java-interop-logger.c
```

Comment:
The fact that you had to prefix `$(JAVA_INTEROP_PATH)` with `../` strongly suggests that `$(JAVA_INTEROP_PATH)` has the wrong value. :-)

Reply:
This line should also be at the top, before all the other `$(MONO_PATH)` values.

Reply:
`$(JAVA_INTEROP_PATH)` is used twice in that file and the related tools are run in different directories.

It might be cleaner to introduce 2 variables for each use. I think I will add `$(JAVA_INTEROP_INCLUDES)` to avoid prefixing it with `../` here.